commit/core.pyx

```diff
@@ -1,837 +1,843 @@
+
+        Parameters
+        ----------
+        model_name : string
+            The name of the model (must match a class name in "commit.models" module)
+        """
+        # Call the specific model constructor
+        if hasattr(commit.models, model_name ) :
+            self.model = getattr(commit.models,model_name)()
+        else :
+            raise ValueError( 'Model "%s" not recognized' % model_name )
+
+        self.set_config('ATOMS_path', pjoin( self.get_config('study_path'), 'kernels', self.model.id ))
+
+
+    def generate_kernels( self, regenerate = False, lmax = 12, ndirs = 32761 ) :
+        """Generate the high-resolution response functions for each compartment.
+        Dispatch to the proper function, depending on the model.
+
+        Parameters
+        ----------
+        regenerate : boolean
+            Regenerate kernels if they already exist (default : False)
+        lmax : int
+            Maximum SH order to use for the rotation procedure (default : 12)
+        ndirs : int
+            Number of directions on the half of the sphere
+        """
+        if not amico.lut.is_valid(ndirs):
+            raise RuntimeError( 'Unsupported value for ndirs.\nNote: Supported values for ndirs are [500, 1000, 1500, 2000, 2500, 3000, 3500, 4000, 4500, 5000, 5500, 6000, 6500, 7000, 7500, 8000, 8500, 9000, 9500, 10000, 32761 (default)]' )
+        if self.scheme is None :
+            raise RuntimeError( 'Scheme not loaded; call "load_data()" first.' )
+        if self.model is None :
+            raise RuntimeError( 'Model not set; call "set_model()" method first.' )
+
+        # store some values for later use
+        self.set_config('lmax', lmax)
+        self.set_config('ndirs', ndirs)
+        self.model.scheme = self.scheme
+
+        print( '\n-> Simulating with "%s" model:' % self.model.name )
+
+        # check if kernels were already generated
+        tmp = glob.glob( pjoin(self.get_config('ATOMS_path'),'A_*.npy') )
+        if len(tmp)>0 and not regenerate :
+            print( '   [ Kernels already computed. Call "generate_kernels( regenerate=True )" to force regeneration. ]' )
+            return
+
+        # create folder or delete existing files (if any)
+        if not exists( self.get_config('ATOMS_path') ) :
+            makedirs( self.get_config('ATOMS_path') )
+        else :
+            for f in glob.glob( pjoin(self.get_config('ATOMS_path'),'*') ) :
+                remove( f )
+
+        # auxiliary data structures
+        aux = amico.lut.load_precomputed_rotation_matrices( lmax, ndirs )
+        idx_IN, idx_OUT = amico.lut.aux_structures_generate( self.scheme, lmax )
+
+        # Dispatch to the right handler for each model
+        tic = time.time()
+        self.model.generate( self.get_config('ATOMS_path'), aux, idx_IN, idx_OUT, ndirs )
+        print( '   [ %.1f seconds ]' % ( time.time() - tic ) )
+
+
+    def load_kernels( self ) :

```

Comment:
If there are kernels created with the previous version, the function `commit.core.Evaluation.load_kernels()` crashes. 

> ValueError: shapes (2640,2275) and (181,2275) not aligned: 2275 (dim 1) != 181 (dim 0)

This is an error generated in AMICO

> `KERNELS['wmr'][i,:,:] = amico.lut.resample_kernel`